test/e2e/specs/editor/blocks/buttons.spec.js

```diff
@@ -100,6 +100,30 @@ test.describe( 'Buttons', () => {
 		).toBeVisible();
 	} );
 
+	test( 'appends http protocol to links added which are missing a protocol', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Regression: https://github.com/WordPress/gutenberg/issues/34307
+		await editor.insertBlock( { name: 'core/buttons' } );
+		await pageUtils.pressKeyWithModifier( 'primary', 'k' );
+		await expect(
+			page.locator( 'role=combobox[name="URL"i]' )
+		).toBeFocused();
+		await page.keyboard.type( 'example.com' );
+		await page.keyboard.press( 'Enter' );
+
+		// Move to "Edit" and switch back to edit
+		await page.keyboard.press( 'Tab' );
+		await page.keyboard.press( 'Enter' );
+
+		// Check the value of the URL input has had http:// prepended.
+		const urlInput = await page.locator( 'role=combobox[name="URL"i]' );
```

Comment:
locator doesn't return a promise.

```suggestion
		const urlInput = page.locator( 'role=combobox[name="URL"i]' );
```

Nit: You can also hoist the locator and reuse it for line 112.

Reply:
Thanks and ✅